### PR TITLE
parsing:Improved variable naming in autolev parser

### LINKED ISSUE #15187: Improve variable naming in autolev parser
Here is an example output from #15165:

```python
import sympy.physics.mechanics as me
import sympy as sm
import math as m
import numpy as np

frame_c = me.ReferenceFrame('c')
frame_d = me.ReferenceFrame('d')
frame_f = me.ReferenceFrame('f')
fd, dc = me.dynamicsymbols('fd dc')
fdd, dcd = me.dynamicsymbols('fd dc', 1)
fdd2, dcd2 = me.dynamicsymbols('fd dc', 2)
r, l = sm.symbols('r l', real=True)
point_o = me.Point('o')
point_e = me.Point('e')
frame_d.orient(frame_f, 'Axis', [fd, frame_f.x])
frame_c.orient(frame_d, 'Axis', [dc, frame_d.y])
frame_c.set_ang_vel(frame_f, (frame_c.ang_vel_in(frame_f)).express(frame_f))
point_e.set_pos(point_o, r*frame_d.y-l*frame_c.x)
point_e.set_pos(point_o, (point_e.pos_from(point_o)).express(frame_d))
point_e.set_vel(frame_f, ((point_e.pos_from(point_o)).dt(frame_f)).express(frame_d))
point_e.set_acc(frame_f, ((point_e.vel(frame_f)).dt(frame_f)).express(frame_d))
```

Name clashes will vary likely happen with the primary import names: `me, sm, np`. There should at least be changed to `_me, _sm, _np` and even better there should be a check to ensure the Autolev code doesn't contain and reserved words and the properly handles them by adding underscores or some kind of pre/post fix. For example in autolev you might declare `MASS ME` as the mass of body E. This will clash with the output code `me`.

The automatic declaration of the time derivatives is currently:

```python
fd, dc = me.dynamicsymbols('fd dc')
fdd, dcd = me.dynamicsymbols('fd dc', 1)
fdd2, dcd2 = me.dynamicsymbols('fd dc', 2)
```

A simple improvement would be:

```python
fd, dc = me.dynamicsymbols('fd dc')
fd_d, dc_d = me.dynamicsymbols('fd dc', 1)
fd_dd, dc_dd = me.dynamicsymbols('fd dc', 2)
```

Where `_d, _dd, _ddd`, etc is dot, double dot, triple dot, etc.

### PR DESCRIPTION
**Pull request by ayushmankoul**
#### References to other Issues or PRs

fixes #15187 

#### Brief description of what is fixed or changed
Added underscore in prefix of all occurences of `me.`,`sm.`,`np.` and `"d"` and in `str_func(z)`,
changed `"d" + str(z)` to `"_" + "d" * z` and in `writeVariables` function changed 373 to 378 lines with following code in `sympy/parsing/autolev/_listener_autolev_antlr.py`
to avoid name clashes which very likely happen with the primary import names: me, sm, np:
```
if i == 0:
	l.append(k)
if i >= 1:
        l.append(k[:-(i+1)])
```

#### Release Notes

* parsing
   *  Improved variable naming in autolev parser

### CODE REVIEW

**Comment by ayushmankoul** [comment]
@NikhilPappu I have altered the code as asked by you.Please have a review of it.

**Comment by NikhilPappu** [comment]
@ayushmankoul Please do not create duplicate PRs. #15200 was fine. I just asked you to close it and reopen the same one so that Travis tests run again and pass (it failed due to some other reason).

**Comment by ayushmankoul** [comment]
@NikhilPappu Sorry, I misinterpreted your words.

**Comment by ayushmankoul** [comment]
@NikhilPappu Sir should I reopen this PR ?

**Comment by NikhilPappu** [comment]
@ayushmankoul You should be able to reopen the previous one but I guess it is okay to work with this now. Please avoid duplicating PRs in the future. You will be able to rewrite anything in a PR so there is no need to switch.

**Comment by NikhilPappu** [comment]
@moorepants Can you have a look at this?

**Comment by ayushmankoul** [comment]
@NikhilPappu Sure sir I will take care of it in future and sorry for burdening your work rather minimizing it.

**Comment by ayushmankoul** [comment]
Hello @NikhilPappu @moorepants Is anything else supposed to be done on this PR ? Please let me know.

File: sympy/parsing/autolev/test-examples/pydy-example-repo/mass_spring_damper.py [hunk 1]
```diff
@@ -1,31 +0,0 @@
-import sympy.physics.mechanics as me
```
Comment (moorepants) [inline sympy/parsing/autolev/test-examples/pydy-example-repo/mass_spring_damper.py]: Why is this file deleted?

File: sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.py [hunk 2]
```diff
@@ -25,15 +25,15 @@
...
-kane = me.KanesMethod(frame_n, q_ind=[q1,q2], u_ind=[u1, u2], kd_eqs = kd_eqs)
+kane = _me.KanesMethod(frame_n, q_ind=[q1,q2], u_ind=[u1, u2], kd_eqs = kd_eqs)
 fr, frstar = kane.kanes_equations([particle_p, particle_r], forceList)
 zero = fr+frstar
 from pydy.system import System
 sys = System(kane, constants = {l:1, m:1, g:9.81},
 specifieds={},
 initial_conditions={q1:.1, q2:.2, u1:0, u2:0},
-times = np.linspace(0.0, 10, 10/.01))
+times = _np.linspace(0.0, 10, 10/.01))
```
Comment (moorepants) [inline sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.py]: All changes of variable names in this file look good.

File: sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.py [hunk 3]
```diff
@@ -25,15 +25,15 @@
 particle_r.mass = m
 force_p = particle_p.mass*(g*frame_n.x)
 force_r = particle_r.mass*(g*frame_n.x)
-kd_eqs = [q1d - u1, q2d - u2]
+kd_eqs = [q1_d - u1, q2_d - u2]
 forceList = [(particle_p.point,particle_p.mass*(g*frame_n.x)), (particle_r.point,particle_r.mass*(g*frame_n.x))]
-kane = me.KanesMethod(frame_n, q_ind=[q1,q2], u_ind=[u1, u2], kd_eqs = kd_eqs)
+kane = _me.KanesMethod(frame_n, q_ind=[q1,q2], u_ind=[u1, u2], kd_eqs = kd_eqs)
```
Comment (moorepants) [inline sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.py]: There should not be spaces around equals signs for kwargs, see PEP8.

File: sympy/parsing/autolev/test-examples/ruletest1.py [hunk 4]
```diff
@@ -1,15 +0,0 @@
-import sympy.physics.mechanics as me
```
Comment (moorepants) [inline sympy/parsing/autolev/test-examples/ruletest1.py]: Why deleted?

File: sympy/parsing/autolev/test-examples/ruletest8.py [hunk 5]
```diff
@@ -1,45 +1,45 @@
...
-a=me.inertia_of_point_mass(particle_p1.mass, particle_p1.point.pos_from(point_o), frame_a) + body_r.inertia[0] + me.inertia_of_point_mass(body_r.mass, body_r.masscenter.pos_from(point_o), frame_a)
-a=body_r.inertia[0] + me.inertia_of_point_mass(body_r.mass, body_r.masscenter.pos_from(point_o), frame_a)
+a=_me.inertia_of_point_mass(particle_p1.mass, particle_p1.point.pos_from(point_o), frame_a)
+a=_me.inertia_of_point_mass(particle_p2.mass, particle_p2.point.pos_from(point_o), frame_a)
+a=body_r.inertia[0] + _me.inertia_of_point_mass(body_r.mass, body_r.masscenter.pos_from(point_o), frame_a)
+a=_me.inertia_of_point_mass(particle_p1.mass, particle_p1.point.pos_from(point_o), frame_a) + _me.inertia_of_point_mass(particle_p2.mass, particle_p2.point.pos_from(point_o), frame_a) + body_r.inertia[0] + _me.inertia_of_point_mass(body_r.mass, body_r.masscenter.pos_from(point_o), frame_a)
+a=_me.inertia_of_point_mass(particle_p1.mass, particle_p1.point.pos_from(point_o), frame_a) + body_r.inertia[0] + _me.inertia_of_point_mass(body_r.mass, body_r.masscenter.pos_from(point_o), frame_a)
+a=body_r.inertia[0] + _me.inertia_of_point_mass(body_r.mass, body_r.masscenter.pos_from(point_o), frame_a)
```
Comment (moorepants) [inline sympy/parsing/autolev/test-examples/ruletest8.py]: There should be space around the equals that is used to define a variable, see PEP8.

File: sympy/parsing/autolev/test-examples/ruletest9.py [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
+particle_p1 = _me.Particle('p1', _me.Point('p1_pt'), _sm.Symbol('m'))
+particle_p2 = _me.Particle('p2', _me.Point('p2_pt'), _sm.Symbol('m'))
+c1, c2, c3 = _me.dynamicsymbols('c1 c2 c3')
+c1_d, c2_d, c3_d = _me.dynamicsymbols('c1 c2 c3', 1)
+body_r_cm = _me.Point('r_cm')
 body_r_cm.set_vel(frame_n, 0)
-body_r_f = me.ReferenceFrame('r_f')
-body_r = me.RigidBody('r', body_r_cm, body_r_f, sm.symbols('m'), (me.outer(body_r_f.x,body_r_f.x),body_r_cm))
+body_r_f = _me.ReferenceFrame('r_f')
+body_r = _me.RigidBody('r', body_r_cm, body_r_f, _sm.symbols('m'), (_me.outer(body_r_f.x,body_r_f.x),body_r_cm))
```
Comment (moorepants) [inline sympy/parsing/autolev/test-examples/ruletest9.py]: space after second to last comma needed

**Review by moorepants — CHANGES_REQUESTED** [review]
LGTM, a few questions to address though.

File: sympy/parsing/autolev/test-examples/ruletest9.py [hunk 7]
```diff
@@ -1,53 +1,53 @@
...
+m = _sm.Matrix([[0]])
 m[0] = 2*x
-m = m.row_insert(m.shape[0], sm.Matrix([[0]]))
+m = m.row_insert(m.shape[0], _sm.Matrix([[0]]))
 m[m.shape[0]-1] = 2*y
 a = 2*m[0]
-m = sm.Matrix([1,2,3,4,5,6,7,8,9]).reshape(3, 3)
-m[0,1]=5
+m = _sm.Matrix([1,2,3,4,5,6,7,8,9]).reshape(3, 3)
+m[0,1] = 5
```
Comment (NikhilPappu) [inline sympy/parsing/autolev/test-examples/ruletest9.py]: Please note that these files are just the outputs generated by the parser. Changes made here are not useful. The write statements in the parser code in `_listener_autolev_antlr.py` are the ones which should be changed.
@moorepants Seems like people are confused by this. Is there any way to make it clear that these are just example codes? Maybe use .txt files instead of .py?

Comment (ayushmankoul) [reply sympy/parsing/autolev/test-examples/ruletest9.py]: The write statements  in `_listerner_autolve_antlr.py` seems to be properly formatted to give spaces around the `=` .Should I keep these changes in `ruletest9.py` or revert back ?

Comment (NikhilPappu) [reply sympy/parsing/autolev/test-examples/ruletest9.py]: @ayushmankoul 
You will have to take a closer look. The examples wouldn't turn out like this if that was the case. Also, you shouldn't worry about making changes to the example files at all. You just have to update the parser code and run the script to update all the examples before commiting.

Comment (ayushmankoul) [reply sympy/parsing/autolev/test-examples/ruletest9.py]: @NikhilPappu Sir I have updated the test examples.Please have a review of it.

File: sympy/parsing/autolev/test-examples/pydy-example-repo/chaos_pendulum.py [hunk 8]
```diff
@@ -43,7 +43,7 @@
 force_b = body_b.mass*(g*frame_n.z)
 kd_eqs = [theta_d - omega, phi_d - alpha]
 forceList = [(body_a.masscenter,body_a.mass*(g*frame_n.z)), (body_b.masscenter,body_b.mass*(g*frame_n.z))]
-kane = _me.KanesMethod(frame_n, q_ind=[theta,phi], u_ind=[omega, alpha], kd_eqs=kd_eqs)
+kane = _me.KanesMethod(frame_n, q_ind=[theta,phi], u_ind=[omega, alpha], kd_eqs = kd_eqs)
```
Comment (Abdullahjavednesar) [inline sympy/parsing/autolev/test-examples/pydy-example-repo/chaos_pendulum.py]: There must be space after ','. No space required around `=`, it was correct initially.

**Comment by NikhilPappu** [comment]
@ayushmankoul Please make the formatting changes in the write methods and related strings of `_listener_autolev_antlr`. Directly changing the files in `test-examples` has no effect on the parser.

File: sympy/parsing/autolev/test-examples/ruletest8.py (hunk 5 above)
Comment (ayushmankoul) [reply sympy/parsing/autolev/test-examples/ruletest8.py]: Hello @NikhilPappu I resolved all other formatting issues in `_listener_autolev_antlr` but I am struck while resolving the space issues around variables in `ruletest8.py` and `ruletest9.py` test-examples, am not able to locate the corresponding code which I need to alter in `_listener_autolev_antlr` for these spaces.If you could give me any hint for any variable hopefully I will be able to fix other.Also,we didn't altered `_listener_autolev_antlr` which could have raised such issues ,then why are we facing these formatting issues while updating test examples ?

Comment (ayushmankoul) [reply sympy/parsing/autolev/test-examples/ruletest8.py]: @NikhilPappu I hope you're doing well.I am waiting for your guidance.

Comment (NikhilPappu) [reply sympy/parsing/autolev/test-examples/ruletest8.py]: @ayushmankoul You will need to look at all the strings and write statements in `_listener_autolev_antlr` and put spaces around things like `+` signs and stuff (see PEP8).
You can check how this looks by rewriting the test files and going over them.
I will be quite busy till this month end due to exams so I apologize if I am unable to get back to you immediately.

Comment (NikhilPappu) [reply sympy/parsing/autolev/test-examples/ruletest8.py]: @ayushmankoul You do not need to fix formatting issues in `_listener_autolev_antlr`. Keep in mind that `_listener_autolev_antlr` writes out parser code and it is the one that generates all the `ruletest` files. So you would need to modify the code that `_listener_autolev_antlr` outputs and not the file itself. Also, do not modify the `ruletest` files by themselves as it has no effect on the parser.

Comment (ayushmankoul) [reply sympy/parsing/autolev/test-examples/ruletest8.py]: Thank You @NikhilPappu for responding despite being busy with your exams.I am following your advice to alter the parser code but I am struck while debugging the `_listener_autolev_antlr` to put spaces around variable declarations.I tried to crosscheck almost every string and write statements which could have caused this issue but there is already proper space given in `" = "` .  Please if you can have a look on `_listener_autolev_antlr` to figure out exactly what needs to be altered to overcome this issue.

Comment (ayushmankoul) [reply sympy/parsing/autolev/test-examples/ruletest8.py]: @moorepants Please help me, I am struck with this issue here.

**Comment by moorepants** [comment]
@ayushmankoul What are your plans with this PR? Will you finish it.

**Comment by ayushmankoul** [comment]
Sorry, @moorepants for the delay. I was waiting for your help I am struck  while removing spaces around variables, I tried to update write statements and relevant strings in `_listener_autolev_antlr` to overcome this issue. Please help me.
> @moorepants Please help me, I am struck with this issue here.

**Comment by NikhilPappu** [comment]
@ayushmankoul The best way to go about this would be to look at all the write statements in the code and to inspect the string variables included in each one. You would have to make sure that PEP8 conventions like proper spacing are followed in all these. Perhaps you are looking only at the write statements and missing out on the variables in those statements which could be strings made with ",".join statements for instance.

**Comment by NikhilPappu** [comment]
@ayushmankoul If you are stuck at understanding the internals of any write statement I would be glad to help with the specifics.

**Comment by ayushmankoul** [comment]
Please @NikhilPappu if you can help me understanding `_me.inertia` write statements I tried to debug it but was not able to reach to any conclusion and simultaneously will be able to help to fix other write statements. We are getting the following error in `_me.inertia` statements:


write

**Comment by NikhilPappu** [comment]
@ayushmankoul You shouldn't worry about any errors. The only tests that exist check the generated code with the code in the test files. The test files should be updated using the procedure described before when the code related to the output is changed.

**Comment by ayushmankoul** [comment]
Hello, @NikhilPappu I was trying to modify parser code in `_listener_autolev_antlr` to put spaces around variables. For instance:-
example


In this case, I have updated the code for `_me.inertia_point_of_mass` in `_listener_autolev_antlr` to put space on right side of `=` which is generating these test examples I am not able to understand the following piece of code how to put space on left side of `=`

antlr


Could you please help me out in this ?

**Comment by NikhilPappu** [comment]
@ayushmankoul You should make this fix in `exitRegularAssign`, `exitIndexAssign` and `exitVecAssign` functions. Strings from functions like `inertia_func()` are used in the `self.write` statements of these functions (The values are being stored using `self.setValue` and obtained in the write statements using `self.getValue`).

**Comment by czgdp1807** [comment]
@ayushmankoul Are you still working on this? Please update the PR with requested changes.

**Comment by percygautam** [comment]
@czgdp1807 I don't think @ayushmankoul is going to complete the work on this pull request. Can you please close this one so that new contributors can take this issue.

**Comment by czgdp1807** [comment]
I am attaching `Please take over` label to it, so anyone can continue this PR.
